### PR TITLE
Push binary and Add Debug Logging

### PR DESCRIPTION
**Pull request by skanda890**
Push binary and #48

### CODE REVIEW

**Comment by skanda890** [comment]
I think merge #48 first and then merge this.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
 tests/*.out
 zig-cache/
-zig-out/
```
Comment (shilangyu) [inline .gitignore:3]: Why this change?

Comment (skanda890) [reply .gitignore:3]: What do you mean

Comment (shilangyu) [reply .gitignore:3]: Why was this line removed from gitignore?

**Comment by shilangyu** [comment]
I don't understand why this branch is based on feat/debug-logging. Can you rebase to master? Thanks!

**Comment by skanda890** [comment]
> I don't understand why this branch is based on feat/debug-logging. Can you rebase to master? Thanks!

Huh, how do I do that?

File: .gitignore (hunk 1 above)
Comment (skanda890) [reply .gitignore:3]: Why? Shouldn't it be removed.

**Comment by shilangyu** [comment]
Could you reopen a new PR with your changes please? The base branch changed a bit and I have merged a few things.
